indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -867,6 +857,18 @@ private boolean tryAssignTask(final Task task, final RemoteTaskRunnerWorkItem ta
     }
   }
 
+  Map<String, ImmutableWorkerInfo> getWorkersEligibleToRunTasks()
```

Comment:
Even though this method was always called under a lock on `workerWithUnacknowledgedTask` before, this change seems OK because it will still be called under the same lock in `tryAssignTask`, and the new caller `getIdleTaskSlotCount` doesn't seem to require locking (as it doesn't update the `workerWithUnacknowledgedTask` map.